### ISSUE TITLE
PunktTrainer fails with ZeroDivisionError on some text.

### ISSUE THREAD

**Issue by bmurr** [nltk/nltk#397]
Running the following code will result in a ZeroDivisionError:

```
from nltk.tokenize.punkt import *
trainer = PunktTrainer()
trainer.train('1.\n2.\n3.')
```

Traceback:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/ben/.virtualenvs/nltk/lib/python2.7/site-packages/nltk/tokenize/punkt.py", line 710, in train
    self.finalize_training(verbose)
  File "/Users/ben/.virtualenvs/nltk/lib/python2.7/site-packages/nltk/tokenize/punkt.py", line 797, in finalize_training
    for (typ1, typ2), ll in self._find_collocations():
  File "/Users/ben/.virtualenvs/nltk/lib/python2.7/site-packages/nltk/tokenize/punkt.py", line 1118, in _find_collocations
    col_count, self._type_fdist.N())
  File "/Users/ben/.virtualenvs/nltk/lib/python2.7/site-packages/nltk/tokenize/punkt.py", line 1058, in _col_log_likelihood
    p2 = 1.0 * (count_b - count_ab) / (N - count_a)
ZeroDivisionError: float division by zero
```

I've also run into a `ValueError` in _col_log_likelihood, when it attempted to do `math.log(0)`.

**Comment by alvations** [comment]
@mrecachinas @bmurr is this fixed by #1247 ?

**Comment by mrecachinas** [comment]
@alvations Correct. There should be no way to get a ZeroDivisionError or ValueError (for `math.log(0)`) with #1247.
